Documentation/coding-guidelines/interop-pinvokes.md

```diff
@@ -0,0 +1,238 @@
+P/Invokes
+=========
+
+This document extends the [Interop Guidelines](interop-guidelines.md) to provide more specific guidelines, notes, and resources for defining P/Invokes.
+
+Guidelines
+----------
+1. Avoid StringBuilder
+2. Use `ExactSpelling` where possible
+3. Only use `SetLastError` where appropriate
+4. Be careful of `BOOL` vs `BOOLEAN`
+5. Watch for buffer size for output strings (null/no null)
+6. Do not create your own buffer cache, use `ArrayPool`
+7. Use blittable types in struct definitions where possible
+
+Attributes
+----------
+
+**Implicit attributes applied to parameter and return values:**
```

Comment:
I think we should leave this sort of stuff out unless there is something prescriptive being offered for our uses in the repo. Our uses of these attributes are really inconsistent, and as far as I can tell, limited only to legacy code inherited from .NET Framework. I don't think any of our Unix shim PInvokes use these, for example.